src/main/java/org/apache/cassandra/sidecar/concurrent/TaskExecutorPool.java

```diff
@@ -204,6 +225,15 @@ Future<Void> closeInternal()
                : workerExecutor.close();
     }
 
+    /**
+     * ThrowableRunnable is similar to {@link Runnable}, but it can throw {@link Exception}
+     */
+    @FunctionalInterface
+    public interface ThrowableRunnable
+    {
+        void run() throws Exception;
```

Comment:
does it make more sense to throw Throwable given the name of the class?
```suggestion
        void run() throws Throwable;
```

Reply:
Sure